crates/subspace-service/src/sync_from_dsn/import_blocks.rs

```diff
@@ -376,3 +380,61 @@ where
 
     Ok(reconstructed_contents.blocks)
 }
+
+#[derive(Clone, Debug)]
+/// Data container to insert the block into the BlockchainDb without checks.
+pub struct RawBlockData<Block: BlockT> {
+    /// Block hash
+    pub hash: Block::Hash,
+    /// Block header
+    pub header: Block::Header,
+    /// Extrinsics of the block
+    pub block_body: Option<Vec<Block::Extrinsic>>,
+    /// Justifications of the block
+    pub justifications: Option<Justifications>,
+}
+
+/// Insert block in the blockchain bypassing checks. Implies that absent block weight is
+/// handled gracefully.
+#[allow(dead_code)] // TODO: remove on usage
+pub fn import_raw_block<B, Block, Client>(
+    client: &Client,
+    raw_block: RawBlockData<Block>,
+) -> Result<(), sp_blockchain::Error>
+where
+    B: backend::Backend<Block>,
+    Block: BlockT,
+    Client: HeaderBackend<Block>
+        + ClientExt<Block, B>
+        + BlockBackend<Block>
+        + ProvideRuntimeApi<Block>
+        + ProofProvider<Block>
+        + LockImportRun<Block, B>
+        + Send
+        + Sync,
+    Client::Api: SubspaceApi<Block, FarmerPublicKey> + ObjectsApi<Block>,
+{
+    let hash = raw_block.hash;
+    let number = *raw_block.header.number();
+    debug!("Importing raw block: {number:?}  - {hash:?} ");
+
+    let mut import_block =
+        BlockImportParams::new(BlockOrigin::NetworkInitialSync, raw_block.header);
+    import_block.justifications = raw_block.justifications;
+    import_block.body = raw_block.block_body;
+    import_block.state_action = StateAction::Skip;
+    import_block.finalized = true;
+    import_block.fork_choice = Some(ForkChoiceStrategy::LongestChain);
+    import_block.import_existing = false;
+
+    let result = client
+        .lock_import_and_run(|operation| client.apply_block(operation, import_block, None))
+        .map_err(|e| {
+            error!("Error during importing of the raw block: {}", e);
+            sp_consensus::Error::ClientImport(e.to_string())
+        })?;
```

Comment:
These 15 lines should really be inlined in a single remaining call to `import_raw_block`. Moreover if you inline `create_raw_block` from https://github.com/subspace/subspace/pull/2763 as well, you'll be able to remove even more code by creating `BlockImportParams` right away instead of creating `RawBlockData` first and converting it into `BlockImportParams` later.